### ISSUE TITLE
Use dropout correctly

### ISSUE THREAD

**Issue by andimarafioti** [andimarafioti/audioContextEncoder#6]
I added a dropout feature to the sequential model. Preliminary tests on it are a bit hard to asses.

I trained two equivalents networks for 800k steps with a learning rate of 1e-3. In orange there's a network with dropout = 0.3 for the linear layer and 0.1 for all conv and deconv layers except the last deconv. In blue is the same network without any dropout. 
I think the sudden change in the orange one in the training SNR comes when I restarted the training with dropout = 0.3 for the linear layer (before it was 0.5, I'm not really sure)

![image](https://user-images.githubusercontent.com/9143109/37088384-73ad9e1e-21fd-11e8-8710-c28b9c4d2e13.png)

![image](https://user-images.githubusercontent.com/9143109/37088489-d1b04700-21fd-11e8-807b-1682997cc947.png)

It seems to work well since the performance on the validation test is better with dropout and worse on the training set.

What do you think? Should I run more tests? Are this parameters good for you? (30% on the linear layer and 10% on convs)

I also tried the same net w/only dropout=50% on convs (blue):

![image](https://user-images.githubusercontent.com/9143109/37088921-2be65c5e-21ff-11e8-8028-882708c63585.png)
